Automattic/wp-calypso
# Signup header banner with refund policy in plans step

Related to https://github.com/Automattic/martech/issues/2790

## Proposed Changes

- Adds a prominent refund policy mention in a header banner on the plans step in signup. This is the last/4th variation in the design mocks: 
- This is behind a feature flag (`onboarding/emphasize-refund-period-in-plans-step`) and only affects the main/default signup flow (so `/start/plans`)

### Media

#### Desktop
<img width="500" alt="Screenshot 2024-04-23 at 11 23 29 AM" src="https://github.com/Automattic/wp-calypso/assets/1705499/26c147d8-1c37-4d0e-a630-f7bea3fee5b2">

#### Mobile (<480px)
<img width="400" alt="Screenshot 2024-04-23 at 11 47 44 AM" src="https://github.com/Automattic/wp-calypso/assets/1705499/3b1321f7-363c-44e3-b098-306d6eca4ad9">

## Testing Instructions



- Go to `/start/plans?flags=onboarding/emphasize-refund-period-in-plans-step`
- Confirm the banner with refund policy is shown
- Confirm back button is clickable and working
- Confirm it is only visible in plans step (so `/start/domains` or checkout should not show the banner)
- With the banner visible, select a plan (paid) and confirm the banner disappears when loading bar is visible and before going to checkout
- Confirm not visible when feature flag is disabled (`/start/plans?flags=-onboarding/emphasize-refund-period-in-plans-step`)

## Pre-merge Checklist
